ts/src/test/tests.ts

```diff
@@ -1388,7 +1388,17 @@ class testMainClass {
                 promises.push (this.testExchangeResponseStatically (exchangeName, exchangeData, testName));
             }
         }
-        await Promise.all (promises);
+        try {
+            await Promise.all (promises);
+        } catch (e) {
+            if (type === 'request') {
+                this.requestTestsFailed = true;
+            } else {
+                this.responseTestsFailed = true;
+            }
+            const errorMessage = '[' + this.lang + '][STATIC_REQUEST]' + '[' + exchange.id + ']' + e.toString ();
+            dump ('[TEST_FAILURE]' + errorMessage);
+        }
```

Comment:
this change was needed. till now, if static tests failed because of throwing exception, then it didn't reach the lines where `this.requestTestsFailed = false` was set, look at this:
https://github.com/ccxt/ccxt/blob/master/ts/src/test/tests.ts#L1165-L1175

so, it also broke promise.all and ended the execution of script without reaching the below lines where we do `exitScript(1)`)`.

Reply:
ok so the issues was not when the tests failed, but when the tests crashed?

Reply:
yes, only when tests crashed, it went in a wild path, where we missed handling it correctly. when it just 'failed', things were handled well.